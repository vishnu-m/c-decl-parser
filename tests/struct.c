/* Ordinary struct */
struct coordinates {
    int x,y;
};

/* Struct with enum declaration inside */
struct rainbow {
    enum Fruit{Violet, Indigo, Blue, Green, Yellow, Red}field1;
};

/* Anonymous struct */
struct {
   char alpha;
   int num;
};

/* Nested Struct */
struct Student {
    char name[50];
    int age;
    struct marks{
        int physics;        
    }m1, m2;
    struct Student *pointer;
    struct Student **double_pointer;
}; 

/*Function pointer */
struct mycallback {
  int (*f)(int);
};

/* Bit fields */
struct bits {
  int x: 5;
  int y: 1;
  int z: 2;
};

/* Forward declaration */
struct context;

struct funcptrs{
  void (*func0)(struct context *ctx);
  void (*func1)(void);
};

struct context{
    struct funcptrs fps;
}; 