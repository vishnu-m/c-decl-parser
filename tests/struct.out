StructDecl	"coordinates"		Parent = "struct.c"
FieldDecl	"x"		type = "int"		Parent = "coordinates"
FieldDecl	"y"		type = "int"		Parent = "coordinates"
StructDecl	"rainbow"		Parent = "struct.c"
EnumDecl	"Fruit"		Parent = "rainbow"
EnumConstantDecl	"Violet"		Parent = "Fruit"
EnumConstantDecl	"Indigo"		Parent = "Fruit"
EnumConstantDecl	"Blue"		Parent = "Fruit"
EnumConstantDecl	"Green"		Parent = "Fruit"
EnumConstantDecl	"Yellow"		Parent = "Fruit"
EnumConstantDecl	"Red"		Parent = "Fruit"
FieldDecl	"field1"		type = "enum Fruit"		Parent = "rainbow"
StructDecl	""		Parent = "struct.c"
FieldDecl	"alpha"		type = "char"		Parent = ""
FieldDecl	"num"		type = "int"		Parent = ""
StructDecl	"Student"		Parent = "struct.c"
FieldDecl	"name"		type = "char [50]"		Parent = "Student"
FieldDecl	"age"		type = "int"		Parent = "Student"
StructDecl	"marks"		Parent = "Student"
FieldDecl	"physics"		type = "int"		Parent = "marks"
FieldDecl	"m1"		type = "struct marks"		Parent = "Student"
FieldDecl	"m2"		type = "struct marks"		Parent = "Student"
FieldDecl	"pointer"		type = "struct Student *"		Parent = "Student"
FieldDecl	"double_pointer"		type = "struct Student **"		Parent = "Student"
StructDecl	"mycallback"		Parent = "struct.c"
FieldDecl	"f"		type = "int (*)(int)"		Parent = "mycallback"
StructDecl	"bits"		Parent = "struct.c"
FieldDecl	"x"		type = "int"		(bitfield = 5)		Parent = "bits"
FieldDecl	"y"		type = "int"		(bitfield = 1)		Parent = "bits"
FieldDecl	"z"		type = "int"		(bitfield = 2)		Parent = "bits"
StructDecl	"context"		Parent = "struct.c"
StructDecl	"funcptrs"		Parent = "struct.c"
FieldDecl	"func0"		type = "void (*)(struct context *)"		Parent = "funcptrs"
FieldDecl	"func1"		type = "void (*)(void)"		Parent = "funcptrs"
StructDecl	"context"		Parent = "struct.c"
FieldDecl	"fps"		type = "struct funcptrs"		Parent = "context"
